src/login/data/reducers.js

```diff
@@ -1,64 +1,68 @@
-import { LOGIN_PERSIST_FORM_DATA, LOGIN_REMOVE_PASSWORD_RESET_BANNER, LOGIN_REQUEST } from './actions';
+import {
+  BACKUP_LOGIN_DATA,
+  DISMISS_PASSWORD_RESET_BANNER,
+  LOGIN_REQUEST,
+} from './actions';
 import { DEFAULT_STATE, PENDING_STATE } from '../../data/constants';
 import { RESET_PASSWORD } from '../../reset-password';
 
 export const defaultState = {
-  loginError: null,
+  loginErrorCode: '',
+  loginErrorContext: {},
   loginResult: {},
-  resetPassword: false,
   loginFormData: {
-    password: '',
-    emailOrUsername: '',
+    formFields: {
+      emailOrUsername: '', password: '',
+    },
     errors: {
-      emailOrUsername: '',
-      password: '',
+      emailOrUsername: '', password: '',
     },
   },
+  shouldBackupState: false,
+  showResetPasswordSuccessBanner: false,
+  submitState: DEFAULT_STATE,
 };
 
 const reducer = (state = defaultState, action = {}) => {
   switch (action.type) {
+    case BACKUP_LOGIN_DATA.BASE:
+      return {
+        ...state,
+        shouldBackupState: true,
+      };
+    case BACKUP_LOGIN_DATA.BEGIN:
```

Comment:
Should not it be `BACKUP_LOGIN_DATA.SUCCESS`?

Reply:
The action `backupLoginFormBegin` has the type BACKUP_LOGIN_DATA.BEGIN. We are initiating a backup request when we call `backupLoginFormBegin`, it hasn't succeeded yet.